pyOCD/target/target_kinetis.py

```diff
@@ -93,26 +93,27 @@ def init(self):
         if isLocked:
             return
 
-        # Prevent the target from resetting if it has invalid code
-        self.transport.writeAP(MDM_CTRL, MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET)
-        while self.transport.readAP(MDM_CTRL) & (MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET) != (MDM_CTRL_DEBUG_REQUEST | MDM_CTRL_CORE_HOLD_RESET):
+        if self.halt_on_connect:
```

Comment:
It does still have issues if there is no valid code. We can continue to improve the capabilities.

Though to be honest, the main reason I added the no halt on connect support was for the older non-interactive version of pyocd-tool. In that version it was required in order for the go, halt, etc commands to work.

But I think this could be a generally useful feature. We'll just need to improve it over time.